### PR TITLE
chore: update to node 18 and alpine 3.17

### PR DESCRIPTION
**Pull request by martabal**
#2242 exists, but seems inactive. This PR:
- updates node version from node 16 to node 18. Some dependencies are updated to be compatible with node 18. `diskusage` dependency is totally incompatible with node 18 and + and so removed. But node 18.15 introduced `statfs` to get disk size/free/available space.
- updates alpine base images from 3.14 to 3.17

Useful links : 

https://nodejs.org/en/blog/announcements/nodejs16-eol
https://www.alpinelinux.org/releases/
https://nodejs.org/api/fs.html#class-fsstatfs

### CODE REVIEW

**Comment by alextran1502** [comment]
the server build seems to not working with the current version of Sharp

File: server/libs/infra/src/repositories/filesystem.provider.ts [hunk 1]
```diff
@@ -69,6 +68,21 @@ export class FilesystemProvider implements IStorageRepository {
...
+
+        const diskUsage: DiskUsage = {
+          available: stats.bavail * stats.bsize,
+          free: stats.bfree * stats.bsize,
+          total: stats.blocks * stats.bsize,
+        };
+
+        resolve(diskUsage);
+      });
+    });
```
Comment (michelheusschen) [inline server/libs/infra/src/repositories/filesystem.provider.ts:86]:
```suggestion
    const stats = await fs.statfs(folder);

    return {
      available: stats.bavail * stats.bsize,
      free: stats.bfree * stats.bsize,
      total: stats.blocks * stats.bsize,
    };
```

**Review by michelheusschen — COMMENTED** [review]
Good job on getting the server image built om arm/v7. Could you please explain the need for `vips-dev` and why version `0.31.3` of sharp is specifically installed instead of the latest version?

**Comment by alextran1502** [comment]
There is an issue with generating thumbnail for HEIC file. I believe it is related to changing the version of `sharp`. I attach a sample file so you can debug

[HEICSAMPLE.zip](https://github.com/immich-app/immich/files/11470375/HEICSAMPLE.zip)

**Comment by alextran1502** [comment]
Another note about support armv7. The team is OK with dropping support for armv7 so if it is the only roadblock to get the dependencies update correctly. Let's remove the architecture from GitHub action

**Comment by michelheusschen** [comment]
Seems like installing `vips-heif` instead of `libheif` fixes the heic thumbnail generation

**Comment by martabal** [comment]
> Another note about support armv7. The team is OK with dropping support for armv7 so if it is the only roadblock to get the dependencies update correctly. Let's remove the architecture from GitHub action

Dropping armv7 is not he only roadblock. I fixed the `Error loading shared library libvips-cpp.so.42: No such file or directory` error by replacing `vips` with `vips-cpp`

https://pkgs.alpinelinux.org/package/edge/community/armhf/vips-cpp

**Review by michelheusschen — APPROVED** [review]
Tested on linux/amd64 with a development + production build and everything still works as expected. The issue Alex found with HEIC images also seems resolved by installing `vips-heif`

**Pull request merged**